docs/using/permissions.md

```diff
@@ -0,0 +1,385 @@
+
+## Permission Types
+
+Wassette supports four types of permissions:
+
+### Storage Permissions
+
+Control file system access for reading and writing files.
+
+**Example uses:**
+- Allow a component to read configuration files
+- Grant write access to output directories
+- Restrict access to specific workspace folders
+
+### Network Permissions
+
+Control outbound network access to specific hosts.
+
+**Example uses:**
+- Allow API calls to external services
+- Permit access to specific domains only
+- Restrict network egress for security
+
+### Environment Variable Permissions
+
+Control access to environment variables.
+
+**Example uses:**
+- Provide API keys to components
+- Share configuration via environment
+- Control access to sensitive credentials
+
+### Memory Permissions
+
+Set memory limits for components (future capability).
+
+**Example uses:**
+- Prevent resource exhaustion
+- Enforce quotas in multi-tenant environments
+
+## Granting Permissions
+
+You can grant permissions in two ways:
+
+### Using Policy Files
+
+Create a `policy.yaml` file alongside your component:
+
+```yaml
+version: "1.0"
+description: "Weather tool permissions"
+permissions:
+  storage:
+    allow:
+      - uri: "fs://workspace/**"
+        access: ["read", "write"]
+      - uri: "fs://config/app.yaml"
+        access: ["read"]
+  network:
+    allow:
+      - host: "api.weather.com"
+      - host: "api.openweathermap.org"
+  environment:
+    allow:
+      - key: "API_KEY"
+      - key: "WEATHER_API_TOKEN"
+```
+
+**Policy file structure:**
+- `version`: Policy format version (currently "1.0")
+- `description`: Human-readable description
+- `permissions`: Permission declarations organized by type
+  - `storage.allow`: List of file system URIs and access types
+  - `network.allow`: List of allowed hosts
+  - `environment.allow`: List of environment variable keys
+
+### Using CLI Commands
+
+Grant permissions at runtime using the `wassette permission grant` command:
+
+**Grant storage access:**
+```bash
+# Read-only access to a directory
+wassette permission grant storage weather-tool fs://workspace/ --access read
+
+# Read and write access
+wassette permission grant storage weather-tool fs://workspace/ --access read,write
+
+# Access to a specific file
+wassette permission grant storage weather-tool fs://config/app.yaml --access read
+```
+
+**Grant network access:**
+```bash
+# Allow access to a specific host
+wassette permission grant network weather-tool api.weather.com
+
+# Allow localhost access
+wassette permission grant network weather-tool localhost:8080
+```
+
+**Grant environment variable access:**
+```bash
+# Grant access to an environment variable
+wassette permission grant environment-variable weather-tool API_KEY
+
+# Grant access to multiple variables
+wassette permission grant environment-variable weather-tool HOME
+wassette permission grant environment-variable weather-tool PATH
+```
+
```

Comment:
the policy file is not a recommended way for granting. You should modify it to use built-in tools and the agent will figure out how to grant permissions. The CLI experience is secondary to that.